install/uninstall.ps1

```diff
@@ -0,0 +1,95 @@
+# Enable a default -Verbose flag for debug output.
+[CmdletBinding()]
+
+# Halt script execution at the first failed command.
+$script:ErrorActionPreference='Stop'
+
+$deleted = $false
+
+function Uninstall {
+    $uninstallDirectory = Get-UninstallDirectory
+    Remove-InstallationFolder $uninstallDirectory
+    Remove-FromPath $uninstallDirectory
+
+    if ($deleted) {
+        Write-Output "Uninstalled AzureAuth!"
+    } else {
+        Write-Output "There were no AzureAuth installations to be deleted."
+    }
+}
+
+function Get-UninstallDirectory {
+    $directory = if (![string]::IsNullOrEmpty($Env:AZUREAUTH_INSTALL_DIRECTORY)) {
+        Write-Error "Uninstalling from custom location is not supported"
+    } 
+    return ([System.IO.Path]::Combine($Env:LOCALAPPDATA, "Programs", "AzureAuth"))
+}
+
+function Remove-InstallationFolder {
+    param ([string]$directory)
+
+    if (Test-Path -Path $directory) {
+        Write-Verbose "Removing installations at '${directory}'"
+        Remove-Item -Force -Recurse $directory
+        $script:deleted = $true
+    } else {
+        Write-Verbose "There were no installations found at '${directory}'"
+    }
+}
+
+function Remove-FromPath {
+    param ([string]$uninstallDirectory)
+
+    $registryPath = 'Registry::HKEY_CURRENT_USER\Environment'
+    $currentPath = (Get-ItemProperty -Path $registryPath -Name PATH -ErrorAction SilentlyContinue).Path
+
+    # We try to find any other AzureAuth installations that are not in default location.

```

Comment:
So if there is azureauth in custom location, we do not support uninstalling it. However, we do support uninstalling from custom location if it is listed in PATH?

Reply:
Just had a conversation with Emilio regarding the same :)
We should either modify the error message on line 23 or just uninstall default azureauth installation and it's location from the PATH.

I would suggest the first approach by mentioning something like `Attempting to remove AzureAuth from all the installation locations found`

Reply:
Yeah. I will change "not support uninstalling from custom location" to "cannot uninstall custom location if not listed in PATH".

This is because if there happens to be more than one AzureAuth installed, for example one in default and another one in custom location (both listed in PATH), if we only delete the AzureAuth in the default location, then the AzureAuth in a custom location is used. To the user, it would appear as if it wasn't uninstalled.

Reply:
100% agree with what you said about multiple installations. Awesome you handled it! :D

Reply:
Thanks for the explanation, Emilio. I have some more questions - Wouldn't removing from PATH and not removing from custom location leave the machine in weird state. For example, there is some `AzureAuth` installation folder lying around but now we do not know its location because we removed that from PATH. Would this take away our ability in future to support custom location uninstallation?

Reply:
He is also deleting the folder before removing from the PATH, correct me if I'm wrong, Emilio.

Reply:
We will just be warning of custom locations and not deleting them.